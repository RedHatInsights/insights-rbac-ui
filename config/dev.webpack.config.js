/* global */
const webpack = require('webpack');
const { resolve } = require('path');
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
const config = require('@redhat-cloud-services/frontend-components-config');

const insightsProxy = {
  https: false,
  ...(process.env.BETA && { deployment: 'beta/apps' }),
};

const webpackProxy = {
  deployment: process.env.BETA ? 'beta/apps' : 'apps',
  useProxy: true,
  proxyVerbose: true,
  env: 'stage-beta',
  localChrome: '/home/martin/insights/insights-chrome/build/',
  appUrl: process.env.BETA ? ['/beta/settings/my-user-access', '/beta/settings/rbac'] : ['/settings/my-user-access', '/settings/rbac'],
};

const { config: webpackConfig, plugins } = config({
  rootFolder: resolve(__dirname, '../'),
  debug: true,
  sassPrefix: '.rbac, .my-user-access',
  client: { overlay: false },
  ...(process.env.PROXY ? webpackProxy : insightsProxy),
});

plugins.push(
  require('@redhat-cloud-services/frontend-components-config/federated-modules')({
    root: resolve(__dirname, '../'),
  })
);

plugins.push(
  new webpack.DefinePlugin({
    API_BASE: JSON.stringify('/api/rbac/v1'),
  })
);

module.exports = (env) => {
  env && env.analyze === 'true' && plugins.push(new BundleAnalyzerPlugin());

  return { ...webpackConfig, plugins };
};
