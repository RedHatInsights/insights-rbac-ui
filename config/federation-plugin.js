const { resolve } = require('path');

const moduleFederationPlugin = require('@redhat-cloud-services/frontend-components-config/federated-modules')({
  root: resolve(__dirname, '../'),
});

module.exports = moduleFederationPlugin;
