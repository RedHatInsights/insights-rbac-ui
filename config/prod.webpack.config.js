const webpack = require('webpack');
const { resolve } = require('path');
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
const config = require('@redhat-cloud-services/frontend-components-config');
const { config: webpackConfig, plugins } = config({
  rootFolder: resolve(__dirname, '../'),
  sassPrefix: '.rbac, .my-user-access',
});

plugins.push(
  require('@redhat-cloud-services/frontend-components-config/federated-modules')({
    root: resolve(__dirname, '../'),
  })
);

plugins.push(
  new webpack.DefinePlugin({
    API_BASE: JSON.stringify('/api/rbac/v1'),
  })
);

module.exports = (env) => {
  env && env.analyze === 'true' && plugins.push(new BundleAnalyzerPlugin());

  return { ...webpackConfig, plugins };
};
