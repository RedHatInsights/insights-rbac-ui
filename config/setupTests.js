import { configure, mount, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import mock, { once } from 'xhr-mock';
import fetchMock from 'fetch-mock';

/**
 * mock fetch
 */
import 'whatwg-fetch';
global.fetchMock = fetchMock;

configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;
global.React = React;

/**
 * Setup mock of tests
 */
mock.setup();
global.mockOnce = once;
global.apiClientMock = mock;

/**
 * setup ENV vars
 */
process.env.BASE_PATH = '/api';

export const entitlementsMock = {
  bundle1: {
    is_entitled: false,
    is_trial: false,
  },
  bundle2: {
    is_entitled: true,
    is_trial: false,
  },
};

export const getUserMock = {
  entitlements: { ...entitlementsMock },
  identity: { user: { is_org_admin: true } },
};

// mock insights instance
global.insights = {
  chrome: {
    auth: {
      getUser: () => new Promise((res) => res(getUserMock)),
    },
    appNavClick: () => Promise.resolve(),
    getUserPermissions: () => Promise.resolve([]),
    isBeta: () => true,
  },
};
