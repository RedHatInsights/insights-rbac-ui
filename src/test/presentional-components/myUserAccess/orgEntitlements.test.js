import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MUAOrgEntitlements from '../../../presentational-components/myUserAccess/orgEntitlements';
import { entitlementsMock } from '../../../../config/setupTests';

describe('<MUAOrgEntitlements />', () => {
  it('should render correctly', () => {
    expect(toJson(mount(
        <MUAOrgEntitlements entitlements={ entitlementsMock }/>
    ))).toMatchSnapshot();
  });
});
