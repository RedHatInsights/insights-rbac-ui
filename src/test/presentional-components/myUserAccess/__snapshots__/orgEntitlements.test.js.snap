// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MUAOrgEntitlements /> should render correctly 1`] = `
<MUAOrgEntitlements
  entitlements={
    Object {
      "bundle1": Object {
        "is_entitled": false,
        "is_trial": false,
      },
      "bundle2": Object {
        "is_entitled": true,
        "is_trial": false,
      },
    }
  }
>
  <Grid
    className="ins-l-mua-bundles"
    hasGutter={true}
    lg={4}
    md={6}
    sm={12}
  >
    <div
      className="pf-l-grid pf-m-all-12-col-on-sm pf-m-all-6-col-on-md pf-m-all-4-col-on-lg pf-m-gutter ins-l-mua-bundles"
    />
  </Grid>
</MUAOrgEntitlements>
`;
